### ISSUE TITLE
Implement Reduction in Custom Losses

### ISSUE THREAD

**Issue by ztsv-av** [ztsv-av/spellbook#22]
Tensorflow and keras losses have a parameter called 'reduction'. When we do multi-gpu training, we set the reduction to None.
- For example: loss_object = tf.losses.CategoricalCrossentropy(reduction=tf.keras.losses.Reduction.NONE) 

There are a total of three parameters:
- 'SUM_OVER_BATCH_SIZE' means the loss instance will return the average of the per-sample losses in the batch.
- 'SUM' means the loss instance will return the sum of the per-sample losses in the batch.
- 'None' means the loss instance will return the full array of per-sample losses.

In order to train our models without any errors using custom loss functions, we need to implement this feature in our loss functions.

**Comment by ztsv-av** [comment]
done.
